config/api/src/main/java/org/jboss/windup/config/metadata/RulesetMetadata.java

```diff
@@ -0,0 +1,77 @@
+package org.jboss.windup.config.metadata;
+
+import java.util.List;
+
+import org.jboss.forge.furnace.addons.Addon;
+import org.jboss.windup.config.RuleProvider;
+import org.jboss.windup.config.phase.RulePhase;
+import org.ocpsoft.rewrite.config.Rule;
+
+/**
+ * Each {@link Addon} that contains {@link RuleProvider} implementations should implement this interface, and by doing
+ * so provide some basic metadata about its contents.
+ * 
+ * @author Jess Sightler <jesse.sightler@gmail.com>
+ * @author <a href="mailto:lincolnbaxter@gmail.com">Lincoln Baxter, III</a>
+ */
+public interface RulesetMetadata
```

Comment:
I would specify in the name that this is specific for Addons. I am lost with all these classes/interfaces.